src/Cafe/OS/libs/nsyshid/BackendLibusb.cpp

```diff
@@ -694,22 +694,58 @@ namespace nsyshid::backend::libusb
 			return false;
 		}
 
-		// ToDo: implement this
-#if 0
-		// is this correct? Discarding "ifIndex" seems like a bad idea
-		int ret = libusb_set_configuration(handleLock->getHandle(), protocol);
-		if (ret == 0) {
-			cemuLog_logDebug(LogType::Force,
-							 "nsyshid::DeviceLibusb::setProtocol(): success");
-			return true;
+		struct libusb_config_descriptor* conf = nullptr;
+		libusb_device* dev = libusb_get_device(handleLock->GetHandle());
+		int getConfig = libusb_get_active_config_descriptor(dev, &conf);
+		if (getConfig == LIBUSB_SUCCESS)
+		{
+			for (uint8 i = 0; i < conf->bNumInterfaces; ++i)
+			{
+				int releaseSuccess = libusb_release_interface(handleLock->GetHandle(), i);
+				if (releaseSuccess < LIBUSB_SUCCESS)
+				{
+					cemuLog_logDebug(LogType::Force, "nsyshid::DeviceLibusb::SetProtocol(): Failed to release interface %i", i);
+					return false;
+				}
+			}
+		}
+		else
+		{
+			cemuLog_logDebug(LogType::Force, "nsyshid::DeviceLibusb::SetProtocol(): Failed to Get Active Config");
+		}
+
+		const int setConfig = libusb_set_configuration(handleLock->GetHandle(), protocol);
+		if (setConfig == LIBUSB_SUCCESS)
+		{
+			getConfig = libusb_get_active_config_descriptor(dev, &conf);
```

Comment:
This (in its current form) causes us to lose all references to the struct that `conf` previously pointed to.
Just made a small test program, to confirm that I am not talking utter nonsense, and valgrind agrees with me that the following snippet leaks:
```c++
struct libusb_config_descriptor *conf = nullptr;
libusb_get_active_config_descriptor(dev, &conf);
libusb_get_active_config_descriptor(dev, &conf);
```